grpc/errors.go

```diff
@@ -96,7 +97,9 @@ func unwrapError(err error, md metadata.MD) error {
 					unwrappedErr,
 				)
 			}
-			outErr = (outErr.(*berrors.BoulderError)).WithSubErrors(suberrs)
+			var berr *berrors.BoulderError
+			errors.As(outErr, &berr)
```

Comment:
Anytime we see `errors.As` without something checking its return value, we should worry. If `errors.As` returns false, `berr` will be nil and we'll get a panic.

As you probably noticed, the existing code has the same issue: If `outErr` does not contain a `*berrors.BoulderError`, it will panic. This works out in practice since we happen to know that `outErr` came from `berrors.New`, which we can look at and see returns the right thing. But that's not actually documented, and in any case we can be a little safer by returning an error instead of panicking if our assumptions fail.

So that would look like `if errors.As(...) { outErr = berr.WithSubErrrs(suberrs) } else { return fmt.Errorf(...) }`.

Note that I recommend the full if/else pairing even though we could follow the Go "early return" style and skip the else, if we reordered the branches to be `if !errors.As(...) { return fmt.Errorf(...) }`. In this case, my style sense says that it's more important to follow the normal pattern for `if errors.As` than it is to follow the pattern of early returns. I think this way leaves the ready more confident that `berr` is only used if the unwrap was succesful.